### PR TITLE
Feat/144 [WIP]

### PR DESCRIPTION
**Pull request by qdozaq**
# WORK IN PROGRESS

### CODE REVIEW

File: src/app/page/Request/foodbox/ConfirmStep.jsx [hunk 1]
```diff
@@ -53,16 +60,48 @@ function ConfirmStep({ dispatch, state }) {
...
-    if (isDonationRequest) {
-      //TODO set the funding status to not funded
-      //Send mission to firestore
+  // TODO we should try to create this mission before paying and if payment fails we delete it or something
+  async function confirmRequest() {
+    const { cart } = state;
+    let mission = {
+      type: MissionType.foodbox,
+      // what does this need to be??
+      // status: ???
```
Comment (qdozaq) [inline src/app/page/Request/foodbox/ConfirmStep.jsx:69]: What does the status need to be for a new mission?

File: src/app/model/User.ts [hunk 2]
```diff
@@ -72,34 +69,6 @@ class User extends BaseModel {
     return this.getCollection("users").where("displayName", ">=", value).limit(limit).get();
   }
 
-  /**
-   * Given a mission object creates a new mission in firestore
-   * returns the new mission id
-   * @param {object} mission
-   * @return {string}
-   */
-  async createMission(mission: MissionInterface): Promise<string> {
```
Comment (qdozaq) [inline src/app/model/User.ts:81]: Moved this method to `Mission.ts` as it makes more sense there is actually associated with the correct default data

**Comment by qdozaq** [comment]
Running into an issue where users that have signed in via a social account still need to input their phone number. Should I just handle in another PR?

File: src/app/component/AddressInput/AddressInput.jsx [hunk 3]
```diff
@@ -43,6 +59,8 @@ AddressInput.propTypes = {
   placeholder: PropTypes.string,
   stage: PropTypes.any, // pickUp or dropOff
   setStage: PropTypes.func,
+  error: PropTypes.bool,
+  onClear: PropTypes.func,
```
Comment (qdozaq) [inline src/app/component/AddressInput/AddressInput.jsx:63]: reminder for myself add value here

File: src/app/component/AddressInput/AddressInput.jsx [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
 import PropTypes from "prop-types";
-import React from "react";
+import React, { useRef, useEffect } from "react";
 
 const { ALGOLIA_API_KEY, ALGOLIA_APP_ID } = process.env;
 
 const AddressInput = (props) => {
-  const { placeholder, setStage, stage } = props;
+  const { error, onClear, placeholder, setStage, stage, value } = props;
+  const placesRef = useRef();
```
Comment (qdozaq) [inline src/app/component/AddressInput/AddressInput.jsx:9]: Adding this ref so we can set a default value in the algolia input.
Needed for when a person goes back in their checkout experience to change details

File: src/app/page/Request/foodbox/ConfirmStep.jsx (hunk 1 above)
Comment (utunga) [reply src/app/page/Request/foodbox/ConfirmStep.jsx:69]: new mission should have status 'Unassigned'  if its paid for by recipient it pops straight into 'funded' therefore 'tentative' (but I would do that *after* the payment is confirmed.)

**Comment by utunga** [comment]
> Running into an issue where users that have signed in via a social account still need to input their phone number. Should I just handle in another PR?

FWIW there was a whole story about that and  PR added (and merged) that if someone tries to volunteer for a mission and they haven't got a phone number then they get asked to add their phone number first. 

The reasoning for this is we don't want to stand in the way of people getting excited and 'joining up' but otoh we do need a phone number before they can actually be assigned a mission.

I don't wanna hit them with a 'confirm your phone number' when they sign in..

File: src/app/page/Request/foodbox/DeliveryStep.jsx [hunk 5]
```diff
@@ -1,25 +1,40 @@
...
 import NavigationButtons from "./NavigationButtons";
 import { User } from "../../../model";
 import { useForm } from "../../../hooks";
 
+const RECAPTCHA_ID = "recaptcha_id";
+
 function DeliveryStep({ dispatch, state }) {
   const classes = useStyles();
   const auth = useSelector((state) => state.firebase.auth);
+  console.log("auth", auth);
```
Comment (mat10tng) [inline src/app/page/Request/foodbox/DeliveryStep.jsx:25]: log.

**Pull request merged**